nemo/tron/api.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Optional
+
+from nemo.tron.checkpointing import save_checkpoint
+from nemo.tron.config import ConfigContainer
+from nemo.tron.eval import evaluate_and_print_results
+from nemo.tron.setup import setup
+from nemo.tron.train import _finish_train, train
+from nemo.tron.utils.common_utils import barrier_and_log, print_rank_0
+
+
+def megatron_pretrain(
+    forward_step_func: Callable,
+    full_config: ConfigContainer,  # TODO: leave as is, or accept modular cfgs as well?
```

Comment:
IMO it's better to leave as is so we don't need to update the API signature each time we update/deprecate various config options